python/cudf_polars/tests/test_drop_nulls.py

```diff
@@ -0,0 +1,56 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES.
+# SPDX-License-Identifier: Apache-2.0
+from __future__ import annotations
+
+import pytest
+
+import polars as pl
+
+from cudf_polars.testing.asserts import (
+    assert_gpu_result_equal,
+    assert_ir_translation_raises,
+)
+
+
+@pytest.fixture(
+    params=[
+        [1, 2, 1, 3, 5, None, None],
+        [1.5, 2.5, None, 1.5, 3, float("nan"), 3],
+        [],
+        [None, None],
+        [1, 2, 3, 4, 5],
+    ]
+)
+def null_data(request):
+    is_empty = pl.Series(request.param).dtype == pl.Null
+    return pl.DataFrame(
+        {
+            "a": pl.Series(request.param, dtype=pl.Float64 if is_empty else None),
+            "b": pl.Series(request.param, dtype=pl.Float64 if is_empty else None),
+        }
+    ).lazy()
+
+
+def test_drop_null(null_data):
+    q = null_data.select(pl.col("a").drop_nulls())
+    assert_gpu_result_equal(q)
+
+
+@pytest.mark.parametrize(
+    "value",
+    [0, pl.col("a").mean(), pl.col("b")],
+    ids=["scalar", "aggregation", "column_expression"],
+)
+def test_fill_null(null_data, value):
+    q = null_data.select(pl.col("a").fill_null(value))
+    assert_gpu_result_equal(q)
+
+
+@pytest.mark.parametrize(
+    "strategy", ["forward", "backward", "min", "max", "mean", "zero", "one"]
+)
+def test_fill_null_with_strategy(null_data, strategy):
+    q = null_data.select(pl.col("a").fill_null(strategy=strategy))
```

Comment:
Can you add a test for the limit keyword?
https://docs.pola.rs/api/python/stable/reference/expressions/api/polars.Expr.fill_null.html

Reply:
I think we might not be passing the necessary options across the rust-python boundary again here:

https://github.com/pola-rs/polars/blob/6f8b4785c1b373b7740d9865671d624af325f9ed/py-polars/src/lazyframe/visitor/expr_nodes.rs#L1073

This doesn't seem to affect `strategy` because using that kwarg apparently results in a different expression node in rust ([FillNullWithStrategy](https://github.com/pola-rs/polars/blob/6f8b4785c1b373b7740d9865671d624af325f9ed/py-polars/src/lazyframe/visitor/expr_nodes.rs#L1266)) but we'd have to expose this to selectively fall back when `limit` is passed rather than the current state which I think is ignoring it.

Perhaps we can come up with a quick polars patch to fix this and tack it on to https://github.com/pola-rs/polars/pull/17702 before it goes in 

cc @wence-

Reply:
Actually, the current state may be sufficient, since `limit` requires `strategy={"forward", "backward"}`, so we get a `FillNullWithStrategy` either way which falls back.